projects/plugins/jetpack/extensions/shared/components/product-management-controls/toolbar-control.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * External dependencies
+ */
+import formatCurrency from '@automattic/format-currency';
+
+/**
+ * WordPress dependencies
+ */
+import { BlockControls } from '@wordpress/block-editor';
+import { MenuGroup, MenuItem, ToolbarDropdownMenu } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { store as editPostStore } from '@wordpress/edit-post';
+import { __, sprintf } from '@wordpress/i18n';
+import { update, warning } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import { jetpackMembershipProductsStore } from './store';
+
+function getProductDescription( product ) {
+	const { currency, interval, price } = product;
+	const amount = formatCurrency( parseFloat( price ), currency );
+	switch ( interval ) {
+		case '1 month':
+			return sprintf(
+				// translators: %s: amount
+				__( '%s / month', 'jetpack' ),
+				amount
+			);
+		case '1 year':
+			return sprintf(
+				// translators: %s: amount
+				__( '%s / year', 'jetpack' ),
+				amount
+			);
+		case 'one-time':
+			return amount;
+	}
+	return sprintf(
+		// translators: %s: amount, plan interval
+		__( '%1$s / %2$s', 'jetpack' ),
+		amount,
+		interval
+	);
+}
+
+function Product( { onClose, product, selectedProductId, selectProduct } ) {
+	const { id, title } = product;
+	const isSelected = selectedProductId && selectedProductId === id;
+	const icon = isSelected ? 'yes' : undefined;
+	const productDescription = product ? ' ' + getProductDescription( product ) : null;
+
+	const handleClick = event => {
+		event.preventDefault();
+		selectProduct( product );
+		onClose();
+	};
+
+	return (
+		<MenuItem icon={ icon } onClick={ handleClick } selected={ isSelected } value={ id }>
+			{ title } : { productDescription }
+		</MenuItem>
+	);
+}
+
+function NewProduct( { onClose } ) {
+	const isEditorSidebarOpened = useSelect( select =>
+		select( editPostStore ).isEditorSidebarOpened()
+	);
+	const { openGeneralSidebar } = useDispatch( editPostStore );
+
+	const handleClick = event => {
+		event.preventDefault();
+		// Open the sidebar if not open
+		if ( ! isEditorSidebarOpened ) {
+			openGeneralSidebar( 'edit-post/block' );
```

Comment:
Ooh, excellent catch!
All different editor variations have their own sidebar name.

- Post: `edit-post/block`
- Site: `edit-site/block-inspector`
- Widget: `edit-widgets/block-inspector`

The tricky part is that IIRC there's no unified way to know that: `isEditorSidebarOpened` only exists on the post editor, site and widget editors have to come up with their own creative ways.

I'll open a follow-up issue to this because it's not worth it to investigate it here.

Reply:
https://github.com/Automattic/jetpack/issues/23443